tests/unit/core/mender-artifact_test.cpp

```diff
@@ -0,0 +1,320 @@
+    fs::path        artifact_path;
+    fs::path        script_file_name;
+
+    vector<uint8_t> CreateArtifact(string custom_script = "") {
+        string script;
+        if (custom_script.empty()) {
+            script = R"(#! /bin/sh
+            DIRNAME=$(dirname $0)
+            echo foobar > ${DIRNAME}/testdata
+
+            mender-artifact write module-image --compression none \
+            --type unit-test -n unit-test-artifact --device-type unit-test \
+            -f ${DIRNAME}/testdata -o ${DIRNAME}/unit-test-artifact.mender || exit 1)";
+        } else {
+            script = custom_script;
+        }
+
+        fs::path tmp_dir = fs::temp_directory_path();
+        script_file_name = tmp_dir / "test-script.sh";
+
+        ofstream os(script_file_name.c_str());
+
+        os << script;
+        os.close();
+
+        chmod(script_file_name.c_str(), S_IRUSR | S_IWUSR | S_IXUSR);
+        system(script_file_name.c_str());
+
+        artifact_path = tmp_dir / "unit-test-artifact.mender";
+        ifstream file(artifact_path);
+        return vector<uint8_t>((istreambuf_iterator<char>(file)), istreambuf_iterator<char>());
+    }
+
+    void InitializeData(mender_update_module_t *update_module, mender_deployment_data_t *deployment_data, mender_artifact_download_data_t *mock_download_data) {
+        mender_create_deployment_data("id", "unit-test-artifact", &deployment_data);
+
+        update_module->artifact_type = "unit-test";
+        update_module->callbacks[MENDER_UPDATE_STATE_DOWNLOAD]
+            = [](MENDER_ARG_UNUSED mender_update_state_t state, MENDER_ARG_UNUSED mender_update_state_data_u data) { return MENDER_OK; };
+        mender_update_module_register(update_module);
+
+        mock_download_data->deployment    = deployment_data;
+        mock_download_data->update_module = update_module;
+    }
+
+    mender_update_module_t          *update_module;
+    mender_deployment_data_t        *deployment_data;
+    mender_artifact_download_data_t *mock_download_data;
+
+    void SetUp() {
+        update_module      = (mender_update_module_t *)malloc(sizeof(mender_update_module_t));
+        deployment_data    = (mender_deployment_data_t *)malloc(sizeof(mender_deployment_data_t));
+        mock_download_data = (mender_artifact_download_data_t *)malloc(sizeof(mender_artifact_download_data_t));
+        InitializeData(update_module, deployment_data, mock_download_data);
+        artifact_data = CreateArtifact();
+        data          = (void *)artifact_data.data();
+    }
+    void TearDown() {
+        fs::remove(script_file_name);
+        fs::remove(artifact_path);
+        DESTROY_AND_NULL(mender_delete_deployment_data, mock_download_data->deployment);
+        free(deployment_data);
+        free(mock_download_data);
+        mender_update_module_unregister_all();
+    }
+};
+
+TEST_F(MenderArtifactTest, CreateCtx) {
+    mender_artifact_ctx_t *ctx = mender_artifact_create_ctx(64);
+    EXPECT_NE(ctx, nullptr);
```

Comment:
I'd prefer `NULL` here.

Reply:
The compiler, for some reason, doesn't accept that